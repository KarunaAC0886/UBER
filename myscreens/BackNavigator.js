import { View, Text, TouchableOpacity } from 'react-native'
import React from 'react'
import {FontAwesome5} from '@expo/vector-icons'
import styles from './styles'
import { useNavigation } from '@react-navigation/native'


const BackNavigator = ({titleName,navpath}) => {
    const navigation = useNavigation();
  return (
    <View style={styles.back_act_cont}>
      <TouchableOpacity onPress={()=>{
        navigation.navigate(`${navpath}`);
      }}><FontAwesome5 style={styles.Back_act_cont_leftarrow} name='arrow-left'/></TouchableOpacity>
      <Text style={styles.Back_act_cont_title}>{titleName}</Text>
    </View>
  )
}

export default BackNavigator