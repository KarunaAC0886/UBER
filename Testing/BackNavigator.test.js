import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import BackNavigator from '../myscreens/BackNavigator';
jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5', 
}));


jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));
jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));
describe('BackNavigator', () => {
  it('BackNavigator page renders correctly', () => {
    
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<BackNavigator/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});