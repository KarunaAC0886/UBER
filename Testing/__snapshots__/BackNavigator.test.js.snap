// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackNavigator BackNavigator page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#000000",
      "height": "13%",
      "justifyContent": "space-between",
      "paddingBottom": "5%",
      "paddingHorizontal": "6%",
      "paddingVertical": "2%",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <FontAwesome5
      name="arrow-left"
      style={
        {
          "color": "#ffffff",
          "fontSize": 20,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Roboto",
        "fontSize": 36,
      }
    }
  />
</View>
`;
